docs/docs/cloud/how-tos/stream_messages.md

```diff
@@ -0,0 +1,412 @@
+# How to stream messages from your graph
+
+LangGraph Cloud supports multiple streaming modes. The main ones are:
+
+- `values`: This streaming mode streams back values of the graph. This is the **full state of the graph** after each node is called.
+- `updates`: This streaming mode streams back updates to the graph. This is the **update to the state of the graph** after each node is called.
+- `messages`: This streaming mode streams back messages - both complete messages (at the end of a node) as well as **tokens** for any messages generated inside a node. This mode is primarily meant for powering chat applications.
+
+
+This guide covers `stream_mode="messages"`.
+
+In order to use this mode, the state of the graph you are interacting with MUST have a messages key that is a list of messages.
+Eg, the state should look something like:
+
+=== "Python"
+    ```python
+    from typing import TypedDict, Annotated
+    from langgraph.graph import add_messages
+    from langchain_core.messages import AnyMessage
+
+    class State(TypedDict):
+        messages: Annotated[list[AnyMessage], add_messages]
+    ```
+
+=== "Javascript"
```

Comment:
let's remove this since we're only using Python for LangGraph API. we can add this once we have a JS version. let's just keep the python example